php/Snoopy.class.inc

```diff
@@ -654,9 +658,17 @@ class Snoopy
 			$port = $this->port;
 		}
 		$this->status = 0;
-		if(function_exists("stream_socket_client") && !is_null($this->IP))
+		if(function_exists("stream_socket_client"))
 		{
-			$opts = array('socket' => array('bindto' => $this->IP.':0'));
+			$opts = array('socket' => array());
```

Comment:
PHP 8 allows this value to be null. We should ensure this value is initialized or not create the stream at all.

Reply:
You mean we can't have an empty array? $this->IP is already checked against null. I was testing with PHP 5.6 and it seemed to be fine as well.

Reply:
From what I found from testing, and perusing some of the source code if `bindto` is not specified it should fall back to the "let system choose" methodology. Which is functionally identical to what `fsockopen` was doing previously. I'm not finding any issues in PHP 7 either though if its unspecified.

Perhaps my code sample comment in other section makes most sense though. It will leave all codepaths identical to previous, only change is if `$useIpv4` is enabled. Just want to confirm before making additional changes.